packages/transactions/src/authorization.ts

```diff
@@ -463,7 +484,7 @@ function verifyMultiSig(
       case StacksMessageType.MessageSignature:
         if (field.pubKeyEncoding === PubKeyEncoding.Uncompressed) haveUncompressed = true;
         const { pubKey, nextSigHash } = nextVerification(
-          curSigHash,
+          isNonSequential(condition.hashMode) ? initialSigHash : curSigHash,
```

Comment:
How about this instead:

```typescript
if (isSequential(condition.hashMode)) {
  curSigHash = nextSigHash;
}
```

Think about it this way: In order-independent mode, we just don't update `curSigHash`. It's easier to read than updating `curSigHash`, and then not using it

Reply:
Yeah, good point that's fine by me 👍🏻 I was going for more control flow where hashmode is used, but I like the elimination of unnecessary work for the readers view